framework/src/Volo.Abp.BackgroundWorkers.Quartz/Volo/Abp/BackgroundWorkers/Quartz/QuartzBackgroundWorkerManager.cs

```diff
@@ -16,28 +16,42 @@ public QuartzBackgroundWorkerManager(IScheduler scheduler)
             _scheduler = scheduler;
         }
 
-        public async Task StartAsync(CancellationToken cancellationToken = default)
+        public virtual async Task StartAsync(CancellationToken cancellationToken = default)
         {
             await _scheduler.ResumeAll(cancellationToken);
         }
 
-        public async Task StopAsync(CancellationToken cancellationToken = default)
+        public virtual async Task StopAsync(CancellationToken cancellationToken = default)
         {
             if (!_scheduler.IsShutdown)
             {
                 await _scheduler.PauseAll(cancellationToken);
             }
         }
 
-        public void Add(IBackgroundWorker worker)
+        public virtual void Add(IBackgroundWorker worker)
+        {
+            AsyncHelper.RunSync(() => ReScheduleJobAsync(worker));
+        }
+
+        protected virtual async Task ReScheduleJobAsync(IBackgroundWorker worker)
         {
             if (worker is IQuartzBackgroundWorker quartzWork)
             {
                 Check.NotNull(quartzWork.Trigger, nameof(quartzWork.Trigger));
                 Check.NotNull(quartzWork.JobDetail, nameof(quartzWork.JobDetail));
-
-                AsyncHelper.RunSync(() => _scheduler.ScheduleJob(quartzWork.JobDetail, quartzWork.Trigger));
+                
+                if (await _scheduler.CheckExists(quartzWork.JobDetail.Key))
+                {
+                    await _scheduler.AddJob(quartzWork.JobDetail, true);
+                    await _scheduler.ResumeJob(quartzWork.JobDetail.Key);
```

Comment:
Why do we call this before Rescheduling? What does it do?

The description for the`ResumeJob` method is: "Resume (un-pause) the IJobDetail with the given name."
The description for the`RescheduleJob` method is: "Remove (delete) the ITrigger with the given name, and store the new given one - which must be associated with the same job."

So does this allow the misfire to be handled before rescheduling?

Reply:
1. The `AddJob` method updates job information, including` JobData`, etc ...
2. The `ResumeJob` method makes Quartz check the Job to execute the missed job.
3. The `RescheduleJob` method updates trigger information.

> So does this allow the misfire to be handled before rescheduling?

Yes, Missed jobs will be executed. You can test it.

Reply:
Yes this seems to be resuming misfired jobs well and is updating triggers and job details as expected. Good work!
I modified `ReScheduleJobAsync` slightly to use `await _scheduler.AddJob(quartzWork.JobDetail, true, true);` and this seemed to work well too with RAMJobStore and AdoJobStore (JobStoreTX). This would mean we can avoid having to use `StoreDurability`.

Reply:
I will check it. 

> Is this something you'd be able to implement too?

What do you mean? Can you explain in detail? thanks.